### PR TITLE
dotCMS/core#22323 Relationships will get cleared if you Save a contentlet before they are loaded

### PR DESCRIPTION
**Pull request by rjvelazco**
https://user-images.githubusercontent.com/72418962/176539914-272e4627-d227-4fff-9f46-fb7cc4bce6d0.mov

### CODE REVIEW

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/contentlet_actions_inc.jsp [hunk 1]
```diff
@@ -192,7 +206,7 @@ function editPage(url, languageId) {
 			<% List<WorkflowActionClass> actionlets = APILocator.getWorkflowAPI().findActionClasses(action); %>
 
 
-
+			<!--  Step: 3 - Agregar la clase 'disabled-contentlet-action' cuando existen relaciones -->
```
Comment (hmoreras) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/contentlet_actions_inc.jsp:209]: english.

**Review by hmoreras — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet.jsp [hunk 2]
```diff
@@ -699,7 +705,7 @@ if(!InodeUtils.isSet(inode) && UtilMethods.isSet(sib) && !UtilMethods.isSet(popu
 		      	dijit.byId('populateDialog').hide();
 		     }
 			 dojo.addOnLoad(function () {
-                 dijit.byId('populateDialog').show();
+				ådijit.byId('populateDialog').show();
```
Comment (wezell) [inline dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet.jsp:708]: utf-8 char looks like it slipped in there...

**Review by wezell — COMMENTED** [review]
Can you post a screencap/vid of the new interactivity?

File: dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet.jsp (hunk 2 above)
[the remark this answers is not in the document]
Comment (rjvelazco) [reply dotCMS/src/main/webapp/html/portlet/ext/contentlet/edit_contentlet.jsp:708]: Done!

**Comment by rjvelazco** [comment]
@wezell Here is a video of [how it works](https://www.loom.com/share/4c3db341316f4d45a3f30748eb9a9c8f). 

In the FE, we do another request to get the `relationship fields`, that's why when the form is submitted the relationship data has not been loaded. After talking to @jcastro-dotcms, it looks like the easier solution.

**Comment by wezell** [comment]
> @wezell Here is a video of [how it works](https://www.loom.com/share/4c3db341316f4d45a3f30748eb9a9c8f).
> 
> In the FE, we do another request to get the `relationship fields`, that's why when the form is submitted the relationship data has not been loaded. After talking to @jcastro-dotcms, it looks like the easier solution.

Ah, that is kind of what I was afraid of - it is actively blocking the user from completing their task and could give rise to user frustration.  

How about something like this:

When a user hits one of the workflow actions, we throw up a progress bar/spinner but only call `saveContent` once the relationships are all loaded.  Then once the `saveContent` is done, the  progress bar/spinner gets closed when the workflow action is returned. This way, it just feels like it is taking a little longer to save, rather than a user clicking an nothing happening.

1.  User load content to edit
2. Relationships not loaded....
3. User edits content and hits save and we show the `progress bar/dialog` like normal, but do not run the `saveContentlet`  yet, so it looks like it is saving normally
4. Relationships still not loaded....
5. `progress bar/dialog` still spinning ... 
6. Relationships finally load....
7. once the relationships are all loaded, we call the `saveContentlet` 
8. Once the `saveContentlet` completes, we close the `progress bar/dialog` like normal.

**Review by oidacra — APPROVED** [review]

**Review by alfredo-dotcms — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**